### DISCUSSION TITLE
Unable to push .net framework package to github 422

### DISCUSSION THREAD

**Discussion by lesma** [community/community#27111] [Actions]
<p>I am having trouble trying to push .net framework packages to github</p>
<p>Commands:</p>
<p>To pack: .nuget\NuGet.exe pack “C:\projects\reponame\ABC.Package” -OutputDirectory “C:\projects\reponame\ABC.Package\bin\Release”<br>
To push .nuget\Nuget.exe push ABC.Package\bin\Release\abc.package.13.1.0.nupkg -Source <a href="https://github.com/orgname/repo-name" rel="noopener nofollow ugc">https://github.com/orgname/repo-name</a></p>
<p>Have tried adding in a API key param to the nuget push but that returns a 403 forbidden without it prompts for username and password which seems to get a little further before getting a response of "Response status code does not indicate success: 422 (Unprocessable Entity) "</p>
<p>.nuspec file is:</p>
<pre><code class="lang-auto">&lt;?xml version="1.0"?&gt;
&lt;package xmlns="http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd"&gt;
  &lt;metadata minClientVersion="4.00"&gt;
    &lt;repository type="git" url="https://github.com/orgname/repo-name" /&gt;
    &lt;id&gt;abc.package&lt;/id&gt;
    &lt;version&gt;$version$&lt;/version&gt;
    &lt;title&gt;$title$&lt;/title&gt;
    &lt;authors&gt;abc&lt;/authors&gt;
    &lt;owners&gt;abc&lt;/owners&gt;
    &lt;requireLicenseAcceptance&gt;false&lt;/requireLicenseAcceptance&gt;
    &lt;description&gt;abc.package&lt;/description&gt;
    &lt;releaseNotes&gt;text.&lt;/releaseNotes&gt;
    &lt;copyright&gt;Copyright 2020&lt;/copyright&gt;
    &lt;tags&gt;&lt;/tags&gt;
  &lt;/metadata&gt;
&lt;/package&gt;
</code></pre>
<p>Can anyone help or is this not supported by GitHub?</p>

**Comment by jcansdale** [comment, marked as answer]
<aside class="quote no-group" data-username="lesma" data-post="1" data-topic="182401">
<div class="title">
<div class="quote-controls"></div>
 lesma:</div>
<blockquote>
<p>To pack: .nuget\NuGet.exe pack “C:\projects\reponame\ABC.Package” -OutputDirectory “C:\projects\reponame\ABC.Package\bin\Release”<br>
To push .nuget\Nuget.exe push ABC.Package\bin\Release\abc.package.13.1.0.nupkg -Source <a href="https://github.com/orgname/repo-name" rel="noopener nofollow ugc">https://github.com/orgname/repo-name </a></p>
</blockquote>
</aside>
<p>You would need to use something like this:</p>
<pre><code class="lang-auto">nuget.exe push ABC.Package\bin\Release\abc.package.13.1.0.nupkg -Source https://nuget.pkg.github.com/orgname -ApiKey &lt;PAT with write:packages and repo scope&gt;
</code></pre>
<p>Note the source is <code>https://nuget.pkg.github.com/orgname</code> not a GitHub repo. The associated repo is specified using the <code>repository</code> element of your .nuspec or (more commonly) the <code>RepositoryUrl</code> property of your .csproj.</p>
<p>I hope that helps!</p>

**Comment by lesma** [comment]
<p>Thanks so much, have been pulling my hair out on this for a while.</p>
